Buildenator/Generators/PropertiesStringGenerator.cs

```diff
@@ -235,6 +256,42 @@ private string GenerateAddToMethodDefinition(ITypedSymbol typedSymbol)
 
 	private string CreateAddToMethodName(ITypedSymbol property) => $"AddTo{property.SymbolPascalName}";
 
+	/// <summary>
+	/// Checks if a typed symbol represents a collection whose element type has a builder.
+	/// </summary>
+	private bool IsCollectionWithChildBuilder(ITypedSymbol typedSymbol)
+	{
+		if (!_builder.UseChildBuilders || _entityToBuilderMappings == null)
+			return false;
+			
+		var collectionMetadata = typedSymbol.GetCollectionMetadata();
+		if (collectionMetadata == null)
+			return false;
+		
+		var elementTypeFullName = collectionMetadata.ElementType.ToDisplayString();
+		return _entityToBuilderMappings.ContainsKey(elementTypeFullName);
+	}
+	
+	/// <summary>
+	/// Tries to get the builder name for a collection's element type.
+	/// Returns true if a child builder exists for the element type.
+	/// </summary>
+	private bool TryGetCollectionChildBuilderName(ITypedSymbol typedSymbol, out string childBuilderName, out string elementTypeName)
+	{
+		childBuilderName = string.Empty;
+		elementTypeName = string.Empty;
+		
+		if (_entityToBuilderMappings == null)
+			return false;
+		
+		var collectionMetadata = typedSymbol.GetCollectionMetadata();
+		if (collectionMetadata == null)
+			return false;
+		
+		elementTypeName = collectionMetadata.ElementType.ToDisplayString();
+		return _entityToBuilderMappings.TryGetValue(elementTypeName, out childBuilderName!);
```

Comment:
Instead of checking the builder exist here, make an eager loading to some variable in the CollectionMetadate class.